### PR TITLE
spdlog: use nixpkgs' fmt instead of bundled one

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
Better fix than https://github.com/NixOS/nixpkgs/pull/106387 to gerbera. (and it's always better to use external libs).

###### Things done
This fixes the gerbera build (thus the pull request above is obsolete)

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by drewrisinger — CHANGES_REQUESTED** [review]
* [x] Commits LGTM
* [x] Diff LGTM
* [ ] DOESN'T build via ``nix-review``:

```
https://github.com/NixOS/nixpkgs/pull/106480
13 packages failed to build:
hal-hardware-analyzer lizardfs nheko python37Packages.qasm2image python37Packages.qiskit python37Packages.qiskit-aer python37Packages.qiskit-aqua python37Packages.qiskit-ignis python38Packages.qasm2image python38Packages.qiskit python38Packages.qiskit-aer python38Packages.qiskit-aqua python38Packages.qiskit-ignis

11 packages built:
bear cloudcompare gerbera mtxclient pdal python37Packages.tiledb python38Packages.tiledb spdlog spdlog_0 tiledb waybar
```

I think I'd tried to essentially do this same thing before, but I don't remember what happened to that PR/why it didn't make it to ``master``. Probably some of these build failures.

I'm the maintainer for the ``pythonPackages.qiskit*``, so I can work to get a patch to get those working again.

**Comment by drewrisinger** [comment]
This patch fixes the ``pythonPackages.qiskit*`` builds. ``pythonPackages.qasm2image`` is broken on master.

```
diff --git a/pkgs/development/python-modules/qiskit-aer/default.nix b/pkgs/development/python-modules/qiskit-aer/default.nix
index 57d89b55711..e5eb877b6b2 100644
--- a/pkgs/development/python-modules/qiskit-aer/default.nix
+++ b/pkgs/development/python-modules/qiskit-aer/default.nix
@@ -8,6 +8,7 @@
 , catch2
 , cmake
 , cython
+, fmt
 , muparserx
 , ninja
 , nlohmann_json
@@ -47,6 +48,7 @@ buildPythonPackage rec {
   buildInputs = [
     blas
     catch2
+    fmt
     muparserx
     nlohmann_json
     spdlog
```

**Comment by SCOTT-HAMILTON** [comment]
Don't think I have the processing power to test and build all of them. But seems like it's mostly a matter of adding fmt to the buildInputs (instead of use the one from spdlog)

File: pkgs/applications/science/electronics/hal-hardware-analyzer/default.nix [hunk 1]
```diff
@@ -42,11 +42,11 @@ stdenv.mkDerivation rec {
   # the qt mkDerivation - the latter forcibly overrides this.
   cmakeBuildType = "MinSizeRel";
 
-  meta = {
+  meta = with stdenv.lib {
```
Comment (drewrisinger) [inline pkgs/applications/science/electronics/hal-hardware-analyzer/default.nix:45]:
```suggestion
  meta = with stdenv.lib; {
```
Syntax error prevents the PR from evaluating.

**Comment by Mic92** [comment]
> This patch fixes the `pythonPackages.qiskit*` builds. `pythonPackages.qasm2image` is broken on master.
> 
> ```
> diff --git a/pkgs/development/python-modules/qiskit-aer/default.nix b/pkgs/development/python-modules/qiskit-aer/default.nix
> index 57d89b55711..e5eb877b6b2 100644
> --- a/pkgs/development/python-modules/qiskit-aer/default.nix
> +++ b/pkgs/development/python-modules/qiskit-aer/default.nix
> @@ -8,6 +8,7 @@
>  , catch2
>  , cmake
>  , cython
> +, fmt
>  , muparserx
>  , ninja
>  , nlohmann_json
> @@ -47,6 +48,7 @@ buildPythonPackage rec {
>    buildInputs = [
>      blas
>      catch2
> +    fmt
>      muparserx
>      nlohmann_json
>      spdlog
> ```

Could just apply this patch. I have the resources to test this.

**Comment by drewrisinger** [comment]
> @SCOTT-HAMILTON Could just apply this patch. I have the resources to test this.

@Mic92 he did, but the current PR fails to evaluate, see my syntax comment above.

**Review by drewrisinger — CHANGES_REQUESTED** [review]
Reviewing both as maintainer of ``python3Packages.qiskit*`` & looking at the rest of this PR:
* [x] Diff LGTM
* [ ] Commits need squashed & divided into one per package, probably?
* [ ] Not clean build of ``lizardfs`` via ``nix-review`` (build works on ``master``). This package hasn't had a new release in ~3 years, though the GitHub repo has had recent commits. Maybe an unstable commit would fix this?? cc maintainer @rushmorem 

All except ``lizardfs`` compile now:
```
https://github.com/NixOS/nixpkgs/pull/106480
1 package failed to build:
lizardfs

23 packages built:
bear cloudcompare gerbera hal-hardware-analyzer mtxclient nheko pdal python37Packages.qasm2image python37Packages.qiskit python37Packages.qiskit-aer python37Packages.qiskit-aqua python37Packages.qiskit-ignis python37Packages.tiledb python38Packages.qasm2image python38Packages.qiskit python38Packages.qiskit-aer python38Packages.qiskit-aqua python38Packages.qiskit-ignis python38Packages.tiledb spdlog spdlog_0 tiledb waybar
```

<details>
<summary>Part of LizardFS error log (I can't actually see a specific error):</summary>

```
[ 93%] Built target master
[ 94%] Building CXX object src/metarestore/CMakeFiles/metarestore.dir/__/master/snapshot_task.cc.o
[ 94%] Building CXX object src/metarestore/CMakeFiles/metarestore.dir/__/master/settrashtime_task.cc.o
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h: In member function 'FSNodeDirectory::iterator FSNodeDirectory::find(const HString&)':
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:240:12: warning: implicitly-declared 'constexpr detail::judy_iterator<hstorage::Handle, FSNode*, false>::judy_iterator(const detail::judy_iterator<hstorage::Handle, FSNode*, false>&)' is deprecated [-Wdeprecated-copy]
  240 |     return it;
      |            ^~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:36,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:143:17: note: because 'detail::judy_iterator<hstorage::Handle, FSNode*, false>' has user-provided 'detail::judy_iterator<T1, T2, IsConst>& detail::judy_iterator<T1, T2, IsConst>::operator=(const detail::judy_iterator<T1, T2, IsConst>&) [with T1 = hstorage::Handle; T2 = FSNode*; bool IsConst = false]'
  143 |  judy_iterator &operator=(const judy_iterator &other) {
      |                 ^~~~~~~~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h: In member function 'FSNodeDirectory::iterator FSNodeDirectory::find(const FSNode*)':
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:261:10: warning: implicitly-declared 'constexpr detail::judy_iterator<hstorage::Handle, FSNode*, false>::judy_iterator(const detail::judy_iterator<hstorage::Handle, FSNode*, false>&)' is deprecated [-Wdeprecated-copy]
  261 |   return it;
      |          ^~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:36,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:143:17: note: because 'detail::judy_iterator<hstorage::Handle, FSNode*, false>' has user-provided 'detail::judy_iterator<T1, T2, IsConst>& detail::judy_iterator<T1, T2, IsConst>::operator=(const detail::judy_iterator<T1, T2, IsConst>&) [with T1 = hstorage::Handle; T2 = FSNode*; bool IsConst = false]'
  143 |  judy_iterator &operator=(const judy_iterator &other) {
      |                 ^~~~~~~~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h: In member function 'FSNodeDirectory::const_iterator FSNodeDirectory::find(const FSNode*) const':
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:278:10: warning: implicitly-declared 'constexpr detail::judy_iterator<hstorage::Handle, FSNode*, true>::judy_iterator(const detail::judy_iterator<hstorage::Handle, FSNode*, true>&)' is deprecated [-Wdeprecated-copy]
  278 |   return it;
      |          ^~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:36,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:143:17: note: because 'detail::judy_iterator<hstorage::Handle, FSNode*, true>' has user-provided 'detail::judy_iterator<T1, T2, IsConst>& detail::judy_iterator<T1, T2, IsConst>::operator=(const detail::judy_iterator<T1, T2, IsConst>&) [with T1 = hstorage::Handle; T2 = FSNode*; bool IsConst = true]'
  143 |  judy_iterator &operator=(const judy_iterator &other) {
      |                 ^~~~~~~~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.h:29,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h: In member function 'FSNodeDirectory::iterator FSNodeDirectory::find(const HString&)':
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:240:12: warning: implicitly-declared 'constexpr detail::judy_iterator<hstorage::Handle, FSNode*, false>::judy_iterator(const detail::judy_iterator<hstorage::Handle, FSNode*, false>&)' is deprecated [-Wdeprecated-copy]
  240 |     return it;
      |            ^~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:36,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.h:29,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:143:17: note: because 'detail::judy_iterator<hstorage::Handle, FSNode*, false>' has user-provided 'detail::judy_iterator<T1, T2, IsConst>& detail::judy_iterator<T1, T2, IsConst>::operator=(const detail::judy_iterator<T1, T2, IsConst>&) [with T1 = hstorage::Handle; T2 = FSNode*; bool IsConst = false]'
  143 |  judy_iterator &operator=(const judy_iterator &other) {
      |                 ^~~~~~~~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.h:29,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h: In member function 'FSNodeDirectory::iterator FSNodeDirectory::find(const FSNode*)':
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:261:10: warning: implicitly-declared 'constexpr detail::judy_iterator<hstorage::Handle, FSNode*, false>::judy_iterator(const detail::judy_iterator<hstorage::Handle, FSNode*, false>&)' is deprecated [-Wdeprecated-copy]
  261 |   return it;
      |          ^~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:36,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.h:29,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:143:17: note: because 'detail::judy_iterator<hstorage::Handle, FSNode*, false>' has user-provided 'detail::judy_iterator<T1, T2, IsConst>& detail::judy_iterator<T1, T2, IsConst>::operator=(const detail::judy_iterator<T1, T2, IsConst>&) [with T1 = hstorage::Handle; T2 = FSNode*; bool IsConst = false]'
  143 |  judy_iterator &operator=(const judy_iterator &other) {
      |                 ^~~~~~~~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.h:29,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h: In member function 'FSNodeDirectory::const_iterator FSNodeDirectory::find(const FSNode*) const':
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:278:10: warning: implicitly-declared 'constexpr detail::judy_iterator<hstorage::Handle, FSNode*, true>::judy_iterator(const detail::judy_iterator<hstorage::Handle, FSNode*, true>&)' is deprecated [-Wdeprecated-copy]
  278 |   return it;
      |          ^~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:36,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.h:29,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:143:17: note: because 'detail::judy_iterator<hstorage::Handle, FSNode*, true>' has user-provided 'detail::judy_iterator<T1, T2, IsConst>& detail::judy_iterator<T1, T2, IsConst>::operator=(const detail::judy_iterator<T1, T2, IsConst>&) [with T1 = hstorage::Handle; T2 = FSNode*; bool IsConst = true]'
  143 |  judy_iterator &operator=(const judy_iterator &other) {
      |                 ^~~~~~~~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h: In function 'void fsnodes_update_ctime(FSNode*, uint32_t)':
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:110:29: warning: implicitly-declared 'constexpr detail::judy_iterator<TrashPathKey, hstorage::Handle, false>::judy_iterator(const detail::judy_iterator<TrashPathKey, hstorage::Handle, false>&)' is deprecated [-Wdeprecated-copy]
  110 |    gMetadata->trash.erase(it);
      |                             ^
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:36,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:143:17: note: because 'detail::judy_iterator<TrashPathKey, hstorage::Handle, false>' has user-provided 'detail::judy_iterator<T1, T2, IsConst>& detail::judy_iterator<T1, T2, IsConst>::operator=(const detail::judy_iterator<T1, T2, IsConst>&) [with T1 = TrashPathKey; T2 = hstorage::Handle; bool IsConst = false]'
  143 |  judy_iterator &operator=(const judy_iterator &other) {
      |                 ^~~~~~~~
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:459:22: note:   initializing argument 1 of 'void judy_map<T1, T2>::erase(judy_map<T1, T2>::iterator) [with T1 = TrashPathKey; T2 = hstorage::Handle; judy_map<T1, T2>::iterator = detail::judy_iterator<TrashPathKey, hstorage::Handle, false>]'
  459 |  void erase(iterator it) {
      |             ~~~~~~~~~^~
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.h:29,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h: In function 'void fsnodes_update_ctime(FSNode*, uint32_t)':
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:110:29: warning: implicitly-declared 'constexpr detail::judy_iterator<TrashPathKey, hstorage::Handle, false>::judy_iterator(const detail::judy_iterator<TrashPathKey, hstorage::Handle, false>&)' is deprecated [-Wdeprecated-copy]
  110 |    gMetadata->trash.erase(it);
      |                             ^
In file included from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node_types.h:36,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/filesystem_node.h:23,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.h:29,
                 from /tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/snapshot_task.cc:21:
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:143:17: note: because 'detail::judy_iterator<TrashPathKey, hstorage::Handle, false>' has user-provided 'detail::judy_iterator<T1, T2, IsConst>& detail::judy_iterator<T1, T2, IsConst>::operator=(const detail::judy_iterator<T1, T2, IsConst>&) [with T1 = TrashPathKey; T2 = hstorage::Handle; bool IsConst = false]'
  143 |  judy_iterator &operator=(const judy_iterator &other) {
      |                 ^~~~~~~~
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:459:22: note:   initializing argument 1 of 'void judy_map<T1, T2>::erase(judy_map<T1, T2>::iterator) [with T1 = TrashPathKey; T2 = hstorage::Handle; judy_map<T1, T2>::iterator = detail::judy_iterator<TrashPathKey, hstorage::Handle, false>]'
  459 |  void erase(iterator it) {
      |             ~~~~~~~~~^~
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h: In instantiation of 'judy_map<T1, T2>::size_type judy_map<T1, T2>::erase(const key_type&) [with T1 = TrashPathKey; T2 = hstorage::Handle; judy_map<T1, T2>::size_type = long unsigned int; judy_map<T1, T2>::key_type = TrashPathKey]':
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/master/settrashtime_task.cc:103:42:   required from here
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:454:3: warning: implicitly-declared 'constexpr detail::judy_iterator<TrashPathKey, hstorage::Handle, false>::judy_iterator(const detail::judy_iterator<TrashPathKey, hstorage::Handle, false>&)' is deprecated [-Wdeprecated-copy]
  454 |   erase(it);
      |   ^~~~~
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:143:17: note: because 'detail::judy_iterator<TrashPathKey, hstorage::Handle, false>' has user-provided 'detail::judy_iterator<T1, T2, IsConst>& detail::judy_iterator<T1, T2, IsConst>::operator=(const detail::judy_iterator<T1, T2, IsConst>&) [with T1 = TrashPathKey; T2 = hstorage::Handle; bool IsConst = false]'
  143 |  judy_iterator &operator=(const judy_iterator &other) {
      |                 ^~~~~~~~
/tmp/nix-build-lizardfs-3.12.0.drv-0/source/src/common/judy_map.h:459:22: note:   initializing argument 1 of 'void judy_map<T1, T2>::erase(judy_map<T1, T2>::iterator) [with T1 = TrashPathKey; T2 = hstorage::Handle; judy_map<T1, T2>::iterator = detail::judy_iterator<TrashPathKey, hstorage::Handle, false>]'
  459 |  void erase(iterator it) {
      |             ~~~~~~~~~^~
[ 95%] Linking CXX static library libmetarestore.a
[ 95%] Built target metarestore
make: *** [Makefile:149: all] Error 2
builder for '/nix/store/plf9aiqlsc38nvdzi1yjn0pgws3m6km1-lizardfs-3.12.0.drv' failed with exit code 2
error: build of '/nix/store/plf9aiqlsc38nvdzi1yjn0pgws3m6km1-lizardfs-3.12.0.drv' failed
```
</details>

``nix-store --read-log /nix/store/plf9aiqlsc38nvdzi1yjn0pgws3m6km1-lizardfs-3.12.0.drv | grep -C 3 error``: Looks like some undefined ``fmt`` references

```
--
/nix/store/dkm3gwl0xrx0wrw6zi5x3px3lpgjhlw4-glibc-2.32-dev/include/bits/string_fortified.h:34: undefined reference to `fmt::v7::detail::basic_data<void>::digits'
/nix/store/4g06wim102dhphbrflg7pv26v7ms8i08-binutils-2.31.1/bin/ld: CMakeFiles/lizardfs.dir/file_info.cc.o:/nix/store/dkm3gwl0xrx0wrw6zi5x3px3lpgjhlw4-glibc-2.32-dev/include/bits/string_fortified.h:34: more undefined references to `fmt::v7::detail::basic_data<void>::digits' follow
/nix/store/4g06wim102dhphbrflg7pv26v7ms8i08-binutils-2.31.1/bin/ld: ../common/libmfscommon.a(slogger.cc.o): in function `spdlog::spdlog_ex::spdlog_ex(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)':
/nix/store/c555ihlcmy1r1vyl591bc6yc2666cxp8-spdlog-1.8.1/include/spdlog/common-inl.h:57: undefined reference to `fmt::v7::format_system_error(fmt::v7::detail::buffer<char>&, int, fmt::v7::basic_string_view<char>)'
/nix/store/4g06wim102dhphbrflg7pv26v7ms8i08-binutils-2.31.1/bin/ld: ../common/libmfscommon.a(slogger.cc.o): in function `spdlog::sinks::rotating_file_sink<std::mutex>::calc_filename(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long)':
/nix/store/qc4g8rn6zwy24b8rlc6asyrmzs7mx2v1-fmt-7.1.3-dev/include/fmt/core.h:2076: undefined reference to `fmt::v7::detail::vformat[abi:cxx11](fmt::v7::basic_string_view<char>, fmt::v7::format_args)'
collect2: error: ld returned 1 exit status
make[2]: *** [src/tools/CMakeFiles/lizardfs.dir/build.make:378: src/tools/lizardfs] Error 1
make[1]: *** [CMakeFiles/Makefile2:1444: src/tools/CMakeFiles/lizardfs.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 106480` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.qasm2image</li>
    <li>python38Packages.qasm2image</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>lizardfs</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>bear</li>
    <li>cloudcompare</li>
    <li>gerbera</li>
    <li>hal-hardware-analyzer</li>
    <li>mtxclient</li>
    <li>nheko</li>
    <li>pdal</li>
    <li>python37Packages.qiskit</li>
    <li>python37Packages.qiskit-aer</li>
    <li>python37Packages.qiskit-aqua</li>
    <li>python37Packages.qiskit-ignis</li>
    <li>python37Packages.tiledb</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-aer</li>
    <li>python38Packages.qiskit-aqua</li>
    <li>python38Packages.qiskit-ignis</li>
    <li>python38Packages.tiledb</li>
    <li>spdlog (spdlog_1)</li>
    <li>spdlog_0</li>
    <li>tiledb</li>
    <li>waybar</li>
  </ul>
</details>

**Comment by Mic92** [comment]
@GrahamcOfBorg build lizardfs

**Comment by Mic92** [comment]
lizardfs looks unrelated.

**Comment by drewrisinger** [comment]
@Mic92, I disagree. ``lizardfs`` seems to be related to old ``fmt`` API references, that aren't in the newer, non-bundled version:

See error from above, shortened for readability
> ```
> /nix/store/...-glibc-2.32-dev/include/bits/string_fortified.h:34: undefined reference to `fmt::v7::detail::basic_data<void>::digits'
> /nix/store/...-binutils-2.31.1/bin/ld: CMakeFiles/lizardfs.dir/file_info.cc.o:/nix/store/...-glibc-2.32-dev/include/bits/string_fortified.h:34: more undefined references to `fmt::v7::detail::basic_data<void>::digits' follow
> ```

**Comment by Mic92** [comment]
> @Mic92, I disagree. `lizardfs` seems to be related to old `fmt` API references, that aren't in the newer, non-bundled version:
> 
> See error from above, shortened for readability
> 
> > ```
> > /nix/store/...-glibc-2.32-dev/include/bits/string_fortified.h:34: undefined reference to `fmt::v7::detail::basic_data<void>::digits'
> > /nix/store/...-binutils-2.31.1/bin/ld: CMakeFiles/lizardfs.dir/file_info.cc.o:/nix/store/...-glibc-2.32-dev/include/bits/string_fortified.h:34: more undefined references to `fmt::v7::detail::basic_data<void>::digits' follow
> > ```

oh sorry, do you have an idea how this can be fixed?

**Comment by Mic92** [comment]
> > @Mic92, I disagree. `lizardfs` seems to be related to old `fmt` API references, that aren't in the newer, non-bundled version:
> > See error from above, shortened for readability
> > > ```
> > > /nix/store/...-glibc-2.32-dev/include/bits/string_fortified.h:34: undefined reference to `fmt::v7::detail::basic_data<void>::digits'
> > > /nix/store/...-binutils-2.31.1/bin/ld: CMakeFiles/lizardfs.dir/file_info.cc.o:/nix/store/...-glibc-2.32-dev/include/bits/string_fortified.h:34: more undefined references to `fmt::v7::detail::basic_data<void>::digits' follow
> > > ```
> 
> oh sorry, do you have an idea how this can be fixed?

I am preparing an upgrade to the latest -rc3.

**Comment by Mic92** [comment]
https://github.com/NixOS/nixpkgs/pull/106669

**Pull request merged**